modules/broker-amqp/src/main/java/org/wso2/broker/amqp/Server.java

```diff
@@ -69,23 +73,50 @@
     private EventExecutorGroup ioExecutors;
     private Channel plainServerChannel;
     private Channel sslServerChannel;
+    private boolean isHaEnabled = false;
+    private volatile boolean isStartCalled = false;
+    private volatile boolean isShutdownHookTriggered = false;
+    private AtomicBoolean isActive = new AtomicBoolean(false);
 
-    public Server(StartupContext context) throws Exception {
-        BrokerConfigProvider configProvider = context.getService(BrokerConfigProvider.class);
+    public Server(StartupContext startupContext) throws Exception {
+        BrokerConfigProvider configProvider = startupContext.getService(BrokerConfigProvider.class);
         this.configuration = configProvider
                 .getConfigurationObject(AmqpServerConfiguration.NAMESPACE, AmqpServerConfiguration.class);
-        this.broker = context.getService(Broker.class);
+        this.broker = startupContext.getService(Broker.class);
+        BrokerHaConfiguration brokerHaConfiguration =
+                configProvider.getConfigurationObject(BrokerHaConfiguration.NAMESPACE, BrokerHaConfiguration.class);
```

Comment:
Ideally, HA configurations should only be read by the broker-coordination module. Other modules should infer that information through API.